### PR TITLE
feat: V2 alpha import and export of organizations

### PR DESCRIPTION
**Pull request by stebenz**

### CODE REVIEW

File: docs/docs/guides/api/export-and-import.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+* Global policies
+* IAM members
+* Global IDPs
+* Global second/multi factors
+* Machine keys
+* PAT's
+* Application keys
+
+Which results in that if you want to import, and you have no defined organization-specific custom policies, the experience for your users will not be exactly like in your old instance.
+
```
Comment (hifabienne) [inline docs/docs/guides/api/export-and-import.md:19]:
```suggestion
```suggestion

::note Note that the ressources will be migrated without the event stream. This means that you will not have the audit trail for the imported objects.

```
```

File: internal/query/user_password.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+	UserState domain.UserState
+}
+
+func (q *Queries) GetHumanPassword(ctx context.Context, orgID, userID string) (passwordHash []byte, algorithm string, err error) {
+	if userID == "" {
+		return nil, "", caos_errs.ThrowInvalidArgument(nil, "QUERY-4Mfsf", "Errors.User.UserIDMissing")
+	}
+	existingPassword, err := q.passwordWriteModel(ctx, userID, orgID)
+	if existingPassword.UserState == domain.UserStateUnspecified || existingPassword.UserState == domain.UserStateDeleted {
```
Comment (adlerhurst) [inline internal/query/user_password.go:33]: will panic if existingPassword is nil, catch the error

File: proto/zitadel/admin.proto [hunk 3]
```diff
@@ -2629,6 +2631,29 @@ service AdminService {
...
+        option (google.api.http) = {
+            post: "/import";
+            body: "*"
+        };
+
+        option (zitadel.v1.auth_option) = {
+            permission: "iam.write";
+        };
+    }
+    // Imports data into instance and creates different objects
```
Comment (adlerhurst) [inline proto/zitadel/admin.proto:2646]: wrong comment

File: proto/zitadel/admin.proto [hunk 4]
```diff
@@ -4621,3 +4646,179 @@ message FailedEvent {
...
+    }
+    message GCSInput{
+        string bucket = 1;
+        string serviceaccount_json = 2;
+        string path = 3;
+    }
+
+    oneof data {
+        ImportDataOrg data_orgs = 1;
+         zitadel.v1.v1.ImportDataOrg data_orgsv1 = 2;
```
Comment (adlerhurst) [inline proto/zitadel/admin.proto:4671]:
```suggestion
        zitadel.v1.v1.ImportDataOrg data_orgsv1 = 2;
```

Comment (adlerhurst) [reply proto/zitadel/admin.proto:4671]: and import path looks strange

File: proto/zitadel/admin.proto [hunk 5]
```diff
@@ -4621,3 +4646,179 @@ message FailedEvent {
...
+    string org_id = 1;
+    zitadel.management.v1.AddOrgRequest org = 3;
+    AddCustomDomainPolicyRequest domain_policy = 4;
+    zitadel.management.v1.AddCustomLabelPolicyRequest label_policy = 5;
+    zitadel.management.v1.AddCustomLockoutPolicyRequest lockout_policy = 6;
+    zitadel.management.v1.AddCustomLoginPolicyRequest login_policy = 7;
+    zitadel.management.v1.AddCustomPasswordComplexityPolicyRequest password_complexity_policy = 8;
+    zitadel.management.v1.AddCustomPrivacyPolicyRequest privacy_policy = 9;
+
+    repeated zitadel.v1.v1.DataProject projects = 10;
```
Comment (adlerhurst) [inline proto/zitadel/admin.proto:4696]: these v1.v1 import paths look strange

File: proto/zitadel/admin.proto [hunk 6]
```diff
@@ -4621,3 +4646,179 @@ message FailedEvent {
...
+    repeated zitadel.management.v1.AddProjectRoleRequest project_roles = 11;
+    repeated  zitadel.v1.v1.DataAPIApplication api_apps = 12;
```
Comment (adlerhurst) [inline proto/zitadel/admin.proto:4698]: multiple lines with 2 spaces

File: proto/zitadel/admin.proto [hunk 7]
```diff
@@ -4621,3 +4646,179 @@ message FailedEvent {
...
+
+    repeated zitadel.idp.v1.IDPUserLink user_links = 33;
+    repeated zitadel.org.v1.Domain domains = 34;
+}
+
+message ImportDataResponse{
+    repeated ImportDataError errors = 1;
+    ImportDataSuccess success = 2;
+}
+message ImportDataError{
```
Comment (adlerhurst) [inline proto/zitadel/admin.proto:4734]:
```suggestion
}

message ImportDataError{
```

**Review by hifabienne — APPROVED** [review]

**Pull request merged**